src/MIDI.hpp

```diff
@@ -948,7 +949,7 @@ bool MidiInterface<Transport, Settings, Platform>::parse()
             mMessage.channel = getChannelFromStatusByte(mPendingMessage[0]);
             mMessage.data1   = mPendingMessage[1];
             mMessage.data2   = 0; // Completed new message has 1 data byte
-            mMessage.length  = 1;
+            mMessage.length  = 2; // Shouldn't be 1, send() checks if length > 1 to send data1, thus length needs to be 2
```

Comment:
**issue**: This feels wrong, especially with the two lines above.
**suggesion**: Maybe `send()` is wrong too?

Reply:
I don't see why it's wrong, message size is in fact 2 bytes, status and data1.

mPendingMessageExpectedLength is set with the packet size accounting with status byte, that message object has 2 bytes, status + data1 which is coherent imho.

Reply:
Truth be told, I'm not really understanding why this is in that particular if

https://github.com/FortySevenEffects/arduino_midi_library/blob/7e9222b4634cdc510f7b76e4c356dfb25f69b174/src/MIDI.hpp#L944-L963

mPendingMessageIndex  is 0, and index doesn't change in the meanwhile, at least until a new packet is received which is on next call to parse()

Reply:
ohhhhhh baby!!! First one?
All the best for the 3 of you! Get some rest.

P.S. It builds locally, but with warnings, several in fact, dunno if I have access to CI logs, if I do, i'll take a peek in the tomorrow or next.

Reply:
Second one. Thanks!

I feel like the whole parser needs a good rewrite to be honest.. With the proposed improvements to SysEx, the upcoming filter/map Thru API and the general fact that this has slowly mutated over the last 10-12 years..